src/xcode/ENA/ENA/Source/Client/Client.swift

```diff
@@ -243,4 +346,37 @@ extension Client {
 			completeWith(FetchedDaysAndHours(hours: hoursResult, days: daysResult))
 		}
 	}
+
+	func fetchHours(
+		_ hours: [Int],
+		day: String,
+		completion completeWith: @escaping FetchHoursCompletionHandler
+	) {
+		var errors = [Client.Failure]()
+		var buckets = [Int: SAPDownloadedPackage]()
+		let group = DispatchGroup()
+
+		hours.forEach { hour in
+			group.enter()
+			self.fetchHour(hour, day: day) { result in
+				switch result {
+				case let .success(hourBucket):
+					buckets[hour] = hourBucket
+				case let .failure(error):
+					errors.append(error)
```

Comment:
the errors.append(...) is not thread safe, right?

Reply:
It is not but the completion handlers are all called on the Main Queue... so this should be fine.